shell/platform/darwin/macos/framework/Source/FlutterKeyboardManagerUnittests.mm

```diff
@@ -356,6 +364,40 @@ - (bool)textInputPlugin {
   return true;
 }
 
+- (bool)forwardKeyEventsToSystemWhenComposing {
+  KeyboardTester* tester = [[KeyboardTester alloc] init];
+
+  tester.isComposing = YES;
+  // Send a down event with composing == YES.
+  [tester.manager handleEvent:keyDownEvent(0x50)];
+
+  NSMutableArray<FlutterAsyncKeyCallback>* channelCallbacks =
+      [NSMutableArray<FlutterAsyncKeyCallback> array];
+  NSMutableArray<FlutterAsyncKeyCallback>* embedderCallbacks =
+      [NSMutableArray<FlutterAsyncKeyCallback> array];
+  [tester recordEmbedderCallsTo:embedderCallbacks];
+  [tester recordChannelCallsTo:channelCallbacks];
+  // TextInputPlugin does not claim the event.
+  [tester respondTextInputWith:NO];
```

Comment:
```suggestion
```

Reply:
Or can you create a way for the mocked text input plugin to record the events, just like the embedder and channel ones?

Reply:
sure ideally I want to verify that the text input plugin did get the event.